sidecar/internal/server/server.go

```diff
@@ -69,8 +76,17 @@ func (ss *SidecarServer) RegisterService(svc SidecarService) {
 // Init creates the internal gRPC server, and registers the SidecarServices.
 // and starts gRPC server.
 func (ss *SidecarServer) Start() {
-	// create the gRPC server and register services
-	ss.server = grpc.NewServer()
+	if ss.enableTLS {
+		creds, err := credentials.NewServerTLSFromFile("/etc/tls/tls.crt", "/etc/tls/tls.key")
+		if err != nil {
+			klog.Fatalf("Could not find TLS file: %v", err)
+		}
+		// create the gRPC server and register services
+		ss.server = grpc.NewServer(grpc.UnaryInterceptor(token.AuthorizationInterceptor(ss.client)), grpc.Creds(creds))
+	}
+	if !ss.enableTLS {
+		ss.server = grpc.NewServer(grpc.UnaryInterceptor(token.AuthorizationInterceptor(ss.client)))
```

Comment:
if tls is not enabled why we need to have add `UnaryInterceptor` here?

Reply:
I think this is something we need to decide. Should we disable auth token verification if TLS is disabled?

Reply:
Removing Interceptor when TLS not used.